### PR TITLE
Lower logging level of "Notifying BuildStarted" et.al. send to Jenkins log.

### PR DESCRIPTION
**Pull request by jonsten**
If a user start posting very long review comments using gerrit trigger, then the system log for Jenkins becomes unreadable since the entire comment is posted there. This change fixes this by truncating all but the first and last line of the message (and concatenating them on the same line).

### CODE REVIEW

**Comment by jonsten** [comment]
@TWestling, @GLundh

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public void buildStarted(Run build, TaskListener taskListener,
                         parameterExpander.getBuildStartedCommand(build, taskListener, (ChangeBasedEvent)event, stats);
                 if (command != null) {
                     if (!command.isEmpty()) {
-                        logger.info("Notifying BuildStarted to gerrit: {}", command);
+                        logger.info("Notifying BuildStarted to gerrit: {}", truncateCommandForLog(command));
```
Comment (jonsten) [inline src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java]: As an alternative we could log without the command on info and then log the command on debug (or lower).

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java [hunk 2]
```diff
@@ -129,4 +129,21 @@ public void buildCompleted(MemoryImprint memoryImprint, TaskListener listener) {
...
+     * @param str String to be truncated
+     * @return Truncated string
+     */
+    private static String truncateCommandForLog(String str) {
+        int firstBreak = str.indexOf('\n');
+        int lastBreak = str.lastIndexOf('\n');
+        if (firstBreak == -1) {
+            return str;
+        }
+        return str.substring(firstBreak) + " ... <truncated> ..." + str.substring(lastBreak, str.length());
```
Comment (TWestling) [inline src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java]: Does this work as intended?
line1
line2
line3
would truncate as:
line2
line3 ... <truncated> ... line3

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java (hunk 1 above)
Comment (rsandell) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java]: Yea, this shouldn't be info level at all I think

Comment (jonsten) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java]: Let's go that route, in general gerrit trigger is a very chatty plugin, changing this to debug would improve the situation somewhat :)

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java (hunk 2 above)
Comment (jonsten) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java]: Well this is embarrassing. But let's go with the decreased log level solution instead :)

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java (hunk 1 above)
Comment (jonsten) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java]: Btw. why are we logging to the logger and not to the listener which ends up in the build log?

Comment (rsandell) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java]: I'm not sure I can remember correctly that far back. But I think it started as a debug log to monitor that the plugin was working correctly. then someone might have wanted it as an info. Or it was an info to begin with because changing logger configs are hard and then found the utility of it good to have and left it there. And we didn't start with listening to build started, that was added a bit later and I guess was copied from `onCompleted` where we didn't dare logging if the logger was closed or something.
_TL DR:_ No good reason at all :)

Comment (jonsten) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritNotifier.java]: Thanks for the explanation :)
